india_compliance/gst_india/overrides/transaction.py

```diff
@@ -754,6 +787,45 @@ def is_export_without_payment_of_gst(doc):
     return is_overseas_doc(doc) and not doc.is_export_with_gst
 
 
+def set_reverse_charge_as_per_gst_settings(doc):
+    gst_settings = frappe.get_cached_value(
+        "GST Settings",
+        "GST Settings",
+        ("enable_rcm_for_unregistered_supplier", "rcm_threshold"),
+        as_dict=1,
+    )
+
+    if (
+        not gst_settings.enable_rcm_for_unregistered_supplier
+        or doc.is_reverse_charge
```

Comment:
Is this condition needed? What if the user changes or removes the template but still keeps this enabled? What if I create a PO using API with is_reverse_charge = 1 but without a tax template?